src/twisted/test/test_threadpool.py

```diff
@@ -182,6 +182,9 @@ def test_threadCreationArgumentsCallInThreadWithCallback(self):
 
         # result callback
         def onResult(success, result):
+            # Spin the GC, which should now delete worker and unique if it's
```

Comment:
This is fantastic.  For extra credit, mention that `gc.collect()` has to happen here because the relevant objects are still in the thread's queue.